app/Http/Controllers/ListingController.php

```diff
@@ -259,4 +313,76 @@ public function rent(Request $request){
 
         return redirect()->route('listings.index');
     }
+
+    public function showAdvertiserListings(Request $request){
+        $listings= Listing::where("user_id", Auth::user()->id)
+            ->with("product");
+
+        if ($request->type != null){
+            $listings->where("type", $request->type);
+        }
+
+        if($request->sort != null && $request->has('sort')){
+            $listings = $listings->orderby('price', $request->sort);
+        }
+
+        $listings = $listings->simplePaginate(10);
+        return view("Listings.listing_list", [
+            'listings' => $listings
+        ]);
+    }
+
+    public function uploadCsvFile(Request $request){
+        $file = $request->file('csv_file');
+        $fileContents = file($file->getPathname());
+        foreach ($fileContents as $line) {
+            $getCsv = str_getcsv($line);
+            $data = [];
+
+            foreach ($getCsv as $csvItem) {
+                $data[] = explode(",", $csvItem);
+            }
+
+            $product = Product::create([
+                'product_name' => $data[0][0],
+                'description' => $data[0][1],
```

Comment:
Met bovegenoemde structuur werkt dit ook niet, Je moet [1][0] doen er is namelijk geen 1 index op index 0